Fix broken installation of CrackMapExec

jsf9k opened the pull request:
## 🗣 Description ##

This pull request:
- Pulls in the changes from cisagov/ansible-role-assessment-tool#42
- Leverages the aforementioned changes to fix a bug that is currently causing AMI builds to fail when installing [`CrackMapExec`](https://github.com/Porchetta-Industries/CrackMapExec)
- Modifies the URL from which `CrackMapExec` is installed

Note that we will build new staging and prod COOL AMIs after pulling these changes into #126.

## 💭 Motivation and context ##

- Resolves #124.
- The old URL from which `CrackMapExec` was being installed now redirects to the new one, so we may as well get the code from the true source.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Revert to using the default branch of [cisagov/ansible-role-assessment-tool](https://github.com/cisagov/ansible-role-assessment-tool) once cisagov/ansible-role-assessment-tool#42 is merged.

jsf9k committed:
Install a Rust compiler for CrackMapExec

The CrackMapExec dependency aardwolf requires a Rust compiler, as the
Python package is only available from PyPI.org as a source
distribution (i.e., there is no wheel available for download)

jsf9k committed:
Update the URL from which CrackMapExec is installed

The old URL now redirects to the new one, so we may as well get the
code from the true source.

dav3r reviewed (approved):
Excellent 👍

mcdonnnj commented on src/crackmapexec.yml:
```diff
@@ -1,16 +1,23 @@
   ansible.builtin.package:
     name:
       - libkrb5-dev
+      - rustc
```
Should we just install `cargo` since that's Rust's preferred method for environment set up?

jsf9k replied:
I'm really only interested in adding enough stuff for Python to build the wheel for `aardwolf`.  I could add `cargo`, but it doesn't seem to be necessary to build the wheel.

If the assessors ever requested us to add a Rust tool to this AMI, then I would modify [cisagov/ansible-role-assessment-tool](https://github.com/cisagov/ansible-role-assessment-tool) to support building a Rust project.  That would presumably involve installing `cargo`, and I would reuse that functionality here.  Do you want me to add an issue to that repo to support Rust (and reference this PR)?  Would that be enough to scratch your `cargo` itch?

mcdonnnj replied:
That will work for me. I ran intro troubles where I needed to install `cargo` for a Python package to successfully build its Rust extensions so I just wanted to mention it as an option.

jsf9k replied:
@mcdonnnj, please see:
- cisagov/ansible-role-assessment-tool#41
- cisagov/ansible-role-assessment-tool#42
- fb16595
- 464f5d8

mcdonnnj reviewed (approved):
Makes sense. I have one question for your consideration.

jsf9k committed:
Temporarily use a non-default branch for cisagov/ansible-role-assessment-tool

This change will be reverted once
cisagov/ansible-role-assessment-tool#42 is merged.

jsf9k committed:
Install Rust compiler via cisagov/ansible-role-assessment-tool

cisagov/ansible-role-assessment-tool#42 introduces some new
functionality for building Rust projects.

jsf9k committed:
Revert "Temporarily use a non-default branch for cisagov/ansible-role-assessment-tool"

This reverts commit fb16595226fc1aa12a1a90f2f99574cd9051c486.

The non-default branch that we were using has been merged via
cisagov/ansible-role-assessment-tool#42.
